### PR TITLE
[s] Assimilation oversights

### PR DESCRIPTION
**Pull request by Kierany9**
unlike goof i actually fix my code after i've been paid

Mind Control now works on vessels inside an enemy hive as long as they aren't already being mind control.
Mind Control no longer works on enemy hivemind hosts, actually prevents double mind controls now.
Mindshield implants reveal the host on implanting much like a headrev. For some reason I thought that was the case with all antagonists.
Mindshield implants break mind control if used on a mind controlled vessel and reveal the host's identity to the correct person.
Maybe fixed the bug where silicons can be assimilation objective targets? Dunno what's up with that one honestly.
Hivemind hosts can no longer be assimilation objective targets.

### CODE REVIEW

**Comment by Kierany9** [comment]
still pulling the 2016 branch i see
i need to fix this

**Comment by Kagus** [comment]
> still pulling the 2016 branch i see
> i need to fix this

Not many people know this, but Lord Singuloth is actually a maintainer of /TG/ code... And he just pulls everything.

**Comment by ExcessiveUseOfCobblestone** [comment]
lol.

This game has so much potential i'm looking forward to playing it when the antags aren't being stupid and going the friendly route.

**Comment by Kierany9** [comment]
DNM

Borgs and AI can be targets for assimilation holy shit how did I not notice this

**Comment by ExcessiveUseOfCobblestone** [comment]
sili-psions

**Comment by davethwave** [comment]
What would happen if use mind control on an AI?

**Comment by Kierany9** [comment]
You can't assimilate silicons

**Comment by Borisvanmemes** [comment]
But you can seem to get them as a target to have in your hivemind at the end of the round.

File: code/game/gamemodes/hivemind/objectives.dm [hunk 1]
```diff
@@ -43,6 +43,10 @@
 /datum/objective/hivemind/assimilate
 	explanation_text = "This is a bug. Error:HIVE3"
 
+/datum/objective/hivemind/assimilate/find_target_by_role(role, role_type=0, invert=0)
+	..()
+	return target
```
Comment (vuonojenmustaturska) [inline code/game/gamemodes/hivemind/objectives.dm]: can't you just leave this proc undefined, if you're not using this return value?

**Pull request merged**